controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -172,52 +167,53 @@ public void triggerReadyJobs(LockedApplication application) {
             if (jobStatus.isRunning(jobTimeoutLimit())) continue;
 
             // Collect the subset of next jobs which have not run with the last changes
-            List<JobType> nextToTrigger = new ArrayList<>();
-            for (JobType nextJobType : order.nextAfter(jobType, application)) {
+            // TODO jvenstad: Change to be step-centric.
+            List<JobType> nextJobs = order.nextAfter(jobType, application);
+            for (JobType nextJobType : nextJobs) {
                 JobStatus nextStatus = application.deploymentJobs().jobStatus().get(nextJobType);
                 if (changesAvailable(application, jobStatus, nextStatus) || nextStatus.isHanging(jobTimeoutLimit()))
-                    nextToTrigger.add(nextJobType);
+                    application = triggerAllowParallel(new Triggering(application, nextJobType, false, "Available change in " + jobType.jobName()), nextJobs, false);
             }
-            // Trigger them in parallel
-            application = trigger(nextToTrigger, application, "Available change in " + jobType.jobName());
             controller.applications().store(application);
         }
     }
 
     /**
-     * Returns true if the previous job has completed successfully with a application version and/or  Vespa version
-     * which is newer (different) than the one last completed successfully in next
+     * Trigger a job for an application, if allowed
+     *
+     * @param triggering the triggering to execute, i.e., application, job type and reason
+     * @param force true to disable checks which should normally prevent this triggering from happening
```

Comment:
Missing explanation of `concurrentlyWith` here: `@param concurrentlyWith production jobs that may run concurrently with the job to trigger.`